src/renderer/components/WarningModal/index.tsx

```diff
@@ -1,14 +1,19 @@
-import React from 'react';
```

Comment:
The logic for showing the warning should be outside of the component so we can reuse the warning modal for other use-cases.

We shouldn't make a generic `disableWarning` setting. It should be scoped like `disableExperimentalWarning`.